File: Frontend/autoquizzer/src/DTO/QuestionsPage/QuestionDTO.ts
import { IDifficultyEnum } from "../IDifficultyEnum";
import { QuestionAnswerDTO } from "./QuestionAnswerDTO";

export interface QuestionDTO {
    questionId : number;
    difficulty: IDifficultyEnum;
    description: string;
    subcategoryId: number;
    categoryId: number;
    questionAnswers: QuestionAnswerDTO[];
}

File: Backend/AutoQuizzer/api/Types/DTOs/QuestionDTO.cs
using Types.Enums;

namespace Types.DTOs
{
    public class QuestionDTO
    {
        public int QuestionId { get; set; }
        public int CategoryId { get; set; }
        public int SubcategoryId { get; set; }
        public string Description { get; set; }
        public Difficulty Difficulty { get; set; }
        public int? UserChoise { get; set; }
        public List<AnswerDTO> QuestionAnswers { get; set; }
    }
}
